### PR TITLE
Add Vary: Accept-Encoding in Gzip middleware

### LINKED ISSUE #627: Gzip middleware does not set Vary: Accept-Encoding
I think this should be set for all responses passing through the middleware, because they might be gzipped if the client supplies `Accept-Encoding: gzip`.

### PR DESCRIPTION
**Pull request by mithrandi**
Before submitting your PR, check that you've:

- [x] Bumped the version number

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

Closes #627.

Better late than never? I'm not sure if this is the best way to manipulate the response header fields, let me know if there's a better way.

### CODE REVIEW

**Comment by mithrandi** [comment]
I don't understand the test failures but they don't _seem_ to be this PR's fault?

File: wai-extra/Network/Wai/Middleware/Gzip.hs [hunk 1]
```diff
@@ -103,17 +103,17 @@ gzip set app env sendResponse = app env $ \res ->
                                  in
                                     doesFileExist compressedVersion >>= \x ->
                                        if x
-                                         then (sendResponse $ ResponseFile s (fixHeaders hs) compressedVersion Nothing)
-                                         else (runAction (ResponseFile s hs file Nothing, nextAction))
+                                         then (sendResponse $ ResponseFile s (vary:fixHeaders hs) compressedVersion Nothing)
```
Comment (snoyberg) [inline wai-extra/Network/Wai/Middleware/Gzip.hs]: I'm worried about this approach, given how many parts of the code need to be modified. What about redefining `sendResponse` to add the necessary header, e.g.:

```haskell
gzip set app env sendResponse' = ...
  where
    sendResponse = sendResponse . addVary
```

**Comment by pdedmon** [comment]
This PR and issue #627 can be closed now.

**Comment by snoyberg** [comment]
Thanks!

**Pull request closed without merging**